flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,378 @@
+    except ValueError:
+        log.error(
+            'Failed to convert %s into a valid quality requirement',
+            flexget_quality_req_string)
+        return None
+def get_flexget_qualities(profile, cutoff_only=False):
+    quality_requirements = []
+    if cutoff_only:
+        name = profile['cutoff']['name']
+        quality_req = radarr_quality_to_flexget_quality_req(name)
+        if not quality_req is None:
+            quality_requirements.append(quality_req)
+    else:
+        for quality in profile['items']:
+            if quality['allowed']:
+                name = quality['quality']['name']
+                quality_req = radarr_quality_to_flexget_quality_req(name)
+                if not quality_req is None:
+                    quality_requirements.append(quality_req)
+
+    return quality_requirements
+
+
+
+class RadarrSet(MutableSet):
+    """ Accesses the Radarr movies using the provided the config """
+
+    def __init__(self, config):
+        self.config = config
+        self.service = RadarrAPIService(
+            config['api_key'],
+            config['base_url'],
+            config['port']
+            )
+
+        # Class member used for caching the items to avoid
+        # unnecessary calls to the Radarr API.
+        # We use the self.items property to access it.
+        # Just set this to None again later to invalidate the cache.
+        self._movie_entries = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def discard(self, value):
+        if not value:
+            return
+
+        if not isinstance(value, Entry):
+            raise TypeError()
+
+        matching_entry = self.__find_matching_entry(value)
+        if matching_entry:
+            movie_id = matching_entry['radarr_id']
+            response = self.service.delete_movie(movie_id)
+            log.verbose('Removed movie %s from Radarr', matching_entry['title'])
+            # Clear the cache
+            self._movie_entries = None
+        else:
+            log.debug('Could not find any matching movie to remove for entry %s', value)
+
+    def __ior__(self, other):
+        for entry in other:
+            if not isinstance(entry, Entry):
+                raise TypeError()
+
+            self.add(entry)
+
+    def __contains__(self, entry):
+        if not value:
+            return False
+
+        if not isinstance(value, Entry):
+            raise TypeError()
+
+        matching_entry = self.__find_matching_entry(value)
+
+        return not matching_entry is None
+
+    def add(self, entry):
+
+        if not isinstance(entry, Entry):
+            raise TypeError()
+
+        # The easiest way to add a movie to Radarr is to
+        # first use the lookup API. Using that we will get
+        # a json response which gives us most of the input
+        # we need for the PUT request.
+
+        result = None
```

Comment:
Because if no match is found the API return an empty json response == empty dictionary. Can I check for False instead? With other words, will `if {}:` evaluate False?